### PR TITLE
feat: support installation on non GLIBC systems

### PR DESCRIPTION
**Pull request by tucksaun**
related to #26
such as Alpine Linux commonly used in Docker images

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -90,17 +92,28 @@ download_and_install() {
...
   fi
-  pkgName="@pnpm/${platform}-${arch}"
-  archive_url="https://registry.npmjs.org/${pkgName}/-/${platform}-${arch}-${version}.tgz"
+  if [ "${platform}" = "linuxstatic" ]; then
+    archive_url="https://github.com/pnpm/pnpm/releases/download/v${version}/pnpm-${platform}-${arch}"
+  else
+    pkgName="@pnpm/${platform}-${arch}"
+    archive_url="https://registry.npmjs.org/${pkgName}/-/${platform}-${arch}-${version}.tgz"
+  fi
+
```
Comment (zkochan) [inline install.sh:101]: I guess we may download all tarballs from github. Not just the linux static one.

Comment (tucksaun) [reply install.sh:101]: this is indeed something possible, and this is what I have done in the past on the Symfony CLI installer.
But I didn't want to suggest such a significant change for a first contribution here 😅

Please note though, artifacts released are not tarballs but directly the binaries.

**Review by umireon — CHANGES_REQUESTED** [review]
I don't think PowerShell changes are relevant to this PR's objective.

File: install.sh [hunk 2]
```diff
@@ -46,7 +46,9 @@ detect_platform() {
   platform="$(uname -s | tr '[:upper:]' '[:lower:]')"
 
   case "${platform}" in
-    linux) platform="linux" ;;
+    linux)
+      platform="linux"
+      getconf GNU_LIBC_VERSION >/dev/null 2>&1 || ldd --version >/dev/null 2>&1 || platform="linuxstatic";;
```
Comment (umireon) [inline install.sh:51]: Please define a function to determine the system is glibc-compatible

```suggestion
    linux)
      if is_glibc_system; then
        platform="linux"
      else
        platform="linuxstatic"
      fi
      ;;
```

**Review by umireon — CHANGES_REQUESTED** [review]

**Comment by umireon** [comment]
I suspect that we should separate this PR into two PRs:

- Making the scripts to download the binaries on GitHub directly (Do not use tarball on npm)
- Making the install.sh to download linuxstatic binary when needed

**Comment by tucksaun** [comment]
@umireon see #29

**Comment by tucksaun** [comment]
rebased on top of #29

**Review by umireon — APPROVED** [review]
LGTM after 29 merged

**Comment by umireon** [comment]
@tucksaun Can you add Alpine Linux test cases on GitHub Actions? My idea is using Docker

**Comment by tucksaun** [comment]
@umireon done

**Review by zkochan — APPROVED** [review]

**Pull request merged**